core/crypto/client_impl.go

```diff
@@ -44,31 +46,39 @@ func (client *clientImpl) NewChaincodeDeployTransaction(chaincodeDeploymentSpec
 	}
 
 	// Get next available (not yet used) transaction certificate
-	tCert, err := client.tCertPool.GetNextTCert()
+	tCerts, err := client.tCertPool.GetNextTCerts(1)
 	if err != nil {
 		client.error("Failed getting next transaction certificate [%s].", err.Error())
```

Comment:
Error message "Failed getting next transaction certificate" appears 8 times in this code change. You must make this more unique for each instance it is used. Otherwise, this will be impossible to debug.